ntp-proto/src/packet/mod.rs

```diff
@@ -467,6 +470,22 @@ impl<'a> NtpPacket<'a> {
             mac.serialize(w)?;
         }
 
+        #[cfg(feature = "ntpv5")]
+        if let NtpHeader::V5(v5::NtpHeaderV5 {
+            desired_size: Some(desired_size),
```

Comment:
Having this in the header feels weird, why not pass the length to the call, or specify that the buffer we write to should be the same size as the request packet?